utils/kafka/producer_test.go

```diff
@@ -0,0 +1,90 @@
+package kafka
+
+import (
+	"strings"
+	"testing"
+
+	"context"
+
+	"github.com/Shopify/sarama"
+	"github.com/Shopify/sarama/mocks"
+	"github.com/carousell/Orion/utils/errors"
+)
+
+func TestProducer(t *testing.T) {
+
+	cfg := sarama.Config{}
+	cfg.Producer.Return.Successes = true
+	cfg.Producer.Return.Errors = true
+	ctx := context.Background()
+
+	t.Run("open producer accepts messages", func(t *testing.T) {
+		asyncProducer := mocks.NewAsyncProducer(t, &cfg)
+		asyncProducer.ExpectInputAndSucceed()
+		producer := Producer{
+			asyncProducer: asyncProducer,
+		}
+		producer.Run()
+		defer producer.Close()
+
+		err := producer.Produce(ctx, "test-topic", "test-key", []byte("test-payload"))
+		if err != nil {
+			t.Errorf("expected no error, got %v", err)
```

Comment:
should replace this with `t.Fatalf()` which ends the test immediately while `t.Errorf()` just marks the test for failure but continues forward (similar comment for other tests)